File: app/src/main/java/com/eftimoff/bakingapp/recipedetails/di/RecipeDetailsComponent.java
package com.eftimoff.bakingapp.recipedetails.di;

import com.eftimoff.bakingapp.app.injection.scopes.PerFragment;
import com.eftimoff.bakingapp.recipedetails.view.RecipeDetailsFragment;

import dagger.Subcomponent;

@PerFragment
@Subcomponent(modules = RecipeDetailsModule.class)
public interface RecipeDetailsComponent {

    void inject(RecipeDetailsFragment recipeDetailsFragment);

}

File: app/src/main/java/com/eftimoff/bakingapp/recipedetails/di/RecipeDetailsComponent.kt
package com.eftimoff.bakingapp.recipedetails.di

import com.eftimoff.bakingapp.app.injection.PerFragment
import com.eftimoff.bakingapp.recipedetails.view.RecipeDetailsFragment
import com.eftimoff.bakingapp.recipelist.di.RecipeDetailsModule
import dagger.Subcomponent

@PerFragment
@Subcomponent(modules = arrayOf(RecipeDetailsModule::class))
interface RecipeDetailsComponent {

    fun inject(recipeDetailsFragment: RecipeDetailsFragment)

}